src/py/flwr_example/quickstart_pytorch/mnist.py

```diff
@@ -0,0 +1,434 @@
+# Copyright 2020 Adap GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""PyTorch MNIST image classification.
+
+The code is generally adapted from PyTorch's Basic MNIST Example. 
+The original code can be inspected in the official PyTorch github:
+
+https://github.com/pytorch/examples/blob/master/mnist/main.py
+"""
+
+
+# mypy: ignore-errors
+# pylint: disable=W0223
+
+import timeit
+from collections import OrderedDict
+from typing import Tuple
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch import Tensor, optim
+from torch.optim.lr_scheduler import StepLR
+from torch.utils.data import DataLoader, Dataset, SubsetRandomSampler
+from torchvision import datasets, transforms
+
+import flwr as fl
+
+
+def dataset_partitioner(
+    dataset: torch.utils.data.Dataset,
+    batch_size: int,
+    client_id: int,
+    number_of_clients: int,
+) -> torch.utils.data.DataLoader:
+    """Helper function to partition datasets
+
+    Parameters
+    ----------
+    dataset: torch.utils.data.Dataset
+        Dataset to be partitioned into *number_of_clients* subsets.
+        
+    batch_size: int 
+        Size of mini-batches used by the returned DataLoader.
+        
+    client_id: int 
+        Unique integer used for selecting a specif partition. 
```

Comment:
Small typo: `specific`

Reply:
Resolved in b1559aeb7d39137b1bdc8f9d5ce5a9dc3a8ca43d